src/SSHDebugPS/CommandRunner.cs

```diff
@@ -9,304 +9,304 @@
-            processStartInfo.CreateNoWindow = true;
-        }
+        private StreamWriter _stdInWriter;
+        private StreamReader _stdOutReader;
+        private StreamReader _stdErrReader;
+
+        protected object _lock = new object();
+
+        public event EventHandler<string> OutputReceived;
+        public event EventHandler<int> Closed;
+        public event EventHandler<ErrorOccuredEventArgs> ErrorOccured;
+
+        public LocalCommandRunner(IPipeTransportSettings settings)
+            : this(settings.Command, settings.CommandArgs)
+        { }
 
-        public void Run()
+        public LocalCommandRunner(string command, string commandArgs)
         {
-            _localProcess = new System.Diagnostics.Process();
-            _localProcess.StartInfo = processStartInfo;
-            _localProcess.Exited += OnProcessExited;
-            _localProcess.EnableRaisingEvents = true;
-            _localProcess.Start();
-
-            _stdoutWriter = new StreamWriter(_localProcess.StandardInput.BaseStream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), 4096, leaveOpen: false);
-            _processReader = new StreamReader(_localProcess.StandardOutput.BaseStream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), false, BUFMAX, false);
-            _outputReadTask = System.Threading.Tasks.Task.Run(() => ReadLoop(_processReader, _cancellationSource.Token, (line) => { OutputReceived?.Invoke(this, line); }));
-
-            _localProcess.ErrorDataReceived += OnErrorOutput;
-            _localProcess.Exited += OnProcessExited;
-            _localProcess.BeginErrorReadLine();
+            CreateProcessStartInfo(command, commandArgs);
         }
 
-        private static int BUFMAX = 4096;
-        private void ReadLoop(StreamReader reader, CancellationToken token, Action<string> action)
+        public void Start()
         {
-            try
+            if (_process != null)
             {
-                while (!token.IsCancellationRequested)
+                if (!_process.HasExited)
                 {
-                    char[] buffer = new char[BUFMAX];
-                    Task<string> task = reader.ReadLineAsync();
-                    task.Wait(token);
+                    Debug.Fail("Process is already running.");
 
-                    if (task.Result == null)
-                    {
-                        lock (_lock)
-                        {
-                            if (!_hasExited && _localProcess.HasExited)
-                            {
-                                _hasExited = true;
-                                Closed?.Invoke(this, _localProcess.ExitCode);
-                            }
-                        }
-                        return; // end of stream
-                    }
-
-                    action(task.Result);
+                    throw new InvalidOperationException("Process already running");
                 }
+                CleanUpProcess();
             }
```

Comment:
Personally, I'd replace this with a check that throws an ObjectDisposedException if the object has been disposed - it's unusual to have an object that implements IDisposable but is still usable after disposing.